From: Hilko Bengen <bengen@debian.org>
Date: Tue, 23 Jan 2018 09:31:30 +0100
Subject: launch: direct: Omit locking option for non-file disks

QEMU does not accept options unrecognized by the block driver
in use. Disable locking only for read-only disks that are
file-backed, as that's the only block driver it is supported
with.

Based on 35320dd0edb94d09d231bcded57aa883a5e7c784
---
 lib/launch-direct.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/launch-direct.c b/lib/launch-direct.c
index 9cc153b..92051b9 100644
--- a/lib/launch-direct.c
+++ b/lib/launch-direct.c
@@ -534,7 +534,8 @@ launch_direct (guestfs_h *g, void *datav, const char *arg)
          escaped_file,
          drv->disk_label ? ",serial=" : "",
          drv->disk_label ? drv->disk_label : "",
-         data->qemu_mandatory_locking ? ",file.backing.file.locking=off" : "",
+         (data->qemu_mandatory_locking && drv->src.protocol == drive_protocol_file) ?
+         ",file.backing.file.locking=off" : "",
          i);
     }
 
